engine/classes/Elgg/Database/MetadataTable.php

```diff
@@ -123,7 +123,10 @@ function create($entity_guid, $name, $value, $value_type = '', $owner_guid = 0,
 		}
 	
 		$access_id = (int) $access_id;
-	
+		
+		if (strlen($value) >= 50) {
```

Comment:
Add `const MAX_VALUE_LENGTH = 65535;` and then compare `strlen($value) >= self::MAX_VALUE_LENGTH`